old/lib/LedgerSMB/DBObject/Payment.pm

```diff
@@ -733,11 +732,21 @@ sub post_bulk {
                 $invoice_array =~ s/\}$/,$invoice_subarray\}/;
             }
         }
-        $self->{transactions} = $invoice_array;
-        $self->{source} = $contact->{source};
 
-
-        $self->call_dbmethod(funcname => 'payment_bulk_post');
+        $self->call_dbmethod(
+            funcname => 'payment_bulk_post',
+            args => {
+                transactions => $invoice_array,
+                batch_id => $self->{batch_id},
+                source => $contact->{source},
```

Comment:
While I understand what you are doing here, I would say the intent of `call_dbmethod` is to add database-side "methods" to the object. To that extent, I would expect that we should only provide the `transactions` and `source` keys as additional parameters.

Reply:
Thanks for the feedback. I hadn't appreciated that some parameters can be provided via `args` while others can be taken from object properties. That makes perfect sense, but I'd seen it as an either/or situation. Patch applied...

Reply:
I haven't either, but that's definitely how it should work. If it doesn't then I/we should fix PGObject::Simple.